app/assets/javascripts/services/topology_service.js

```diff
@@ -117,4 +127,30 @@ ManageIQ.angular.app.service('topologyService', function() {
     }
     return kinds
   };
+
+  // this injects some common code in the controller - temporary pending a proper merge
+  this.mixinSearch = function($scope) {
+    var topologyService = this;
+
+    $scope.searching = false;
+    $scope.notFound = false;
+
+    $scope.searchNode = function() {
+      var svg = topologyService.getSVG($scope.d3);
+      var query = $('input#search_topology')[0].value;
```

Comment:
Sorry, not doing that, this has nothing to do with anything angular related, no point in using `angular.element` here.

(This looks out of place in an angular controller, I agree completely, but it is supposed to look out of place, because that's one of those things that's done in a silly way until we make the custom toolbar angular too.)